components/sst_importer/src/sst_writer.rs

```diff
@@ -322,46 +355,62 @@ mod tests {
+            b"rk2"
+        } else {
+            b"k2"
+        };
 
-        // put value
-        let mut pair = Pair::default();
-        pair.set_key(b"k1".to_vec());
-        pair.set_value(b"short_value".to_vec());
-        pairs.push(pair);
-
-        // delete value
-        let mut pair = Pair::default();
-        pair.set_key(b"k2".to_vec());
-        pair.set_op(PairOp::Delete);
-        pairs.push(pair);
-
-        // generate meta
-        batch.set_ttl(10);
-        batch.set_pairs(pairs.into());
-        w.write(batch).unwrap();
-        assert_eq!(w.default_entries, 1);
-        assert_eq!(w.default_deletes, 1);
-        match api_version {
-            ApiVersion::V1ttl => {
-                // ttl takes 8 more bytes
-                assert_eq!(
-                    w.default_bytes as usize,
-                    b"zk1".len() + b"short_value".len() + 8 + b"zk2".len()
-                );
-            }
-            ApiVersion::V2 => {
-                // ttl takes 8 more bytes and meta take 1 more byte
-                assert_eq!(
-                    w.default_bytes as usize,
-                    b"zk1".len() + b"short_value".len() + b"zk2".len() + 9
-                );
+        dispatch_api_version!(api_version, {
+            // put value
+            let mut pair = Pair::default();
+            pair.set_op(PairOp::Put);
+            pair.set_key(key1.to_vec());
+            pair.set_value(b"short_value".to_vec());
+            pairs.push(pair);
+
+            // delete value
+            let mut pair = Pair::default();
+            pair.set_op(PairOp::Delete);
+            pair.set_key(key2.to_vec());
+            pairs.push(pair);
+
+            // generate meta
+            batch.set_ttl(10);
+            batch.set_pairs(pairs.into());
+            w.write(batch).unwrap();
+            assert_eq!(w.default_entries, 1);
+            assert_eq!(w.default_deletes, 1);
+            match api_version {
+                ApiVersion::V1ttl => {
+                    // ttl takes i more bytes
+                    assert_eq!(
+                        w.default_bytes as usize,
+                        b"zk1".len() + b"short_value".len() + 8 + b"zk2".len()
+                    );
+                }
+                ApiVersion::V2 => {
+                    // ttl takes 8 more bytes and meta take 1 more byte
+                    assert_eq!(
+                        w.default_bytes as usize,
+                        (b"z".len() + Key::from_raw(b"rk1").as_encoded().len() + 8)
+                            + (b"short_value".len() + 9)
+                            + (b"z".len() + Key::from_raw(b"rk2").as_encoded().len() + 8)
+                    );
+                }
+                _ => unreachable!(),
             }
-            _ => unreachable!(),
-        }
 
-        let metas = w.finish().unwrap();
-        assert_eq!(metas.len(), 1);
+            let metas = w.finish().unwrap();
```

Comment:
Do we have verification of generated SST somewhere ?

Reply:
The SST's KV pairs are generated by the client as well as the meta.